react/booksManager.js

```diff
@@ -0,0 +1,70 @@
+import {Component} from 'react'
+
+export class BooksManager extends Component {
+  constructor(props) {
+    super(props)
+    this.clearInput()
+  }
+
+  createBook() {
+    const options = {
+      variables: {
+        name: this.state.name,
+        authors: this.state.authors.split(',')
+      },
+      refetchQueries: [{query: this.props.onGet}]
+    }
+
+    this.props.onCreate(options).then(this.clearInput.bind(this))
+  }
+
+  deleteBook() {
+    const options = {
+      variables: {
+        id: this.state.id
+      },
+      refetchQueries: [{query: this.props.onGet}]
+    }
+
+    this.props.onDelete(options).then(this.clearInput.bind(this))
+  }
+
+  clearInput() {
+    this.state = Object.assign({}, {
```

Comment:
This won't work with React. Changing state should always be done via the `setState` function, like this:

```js
this.setState({
  name: '',
  authors: '',
  id: '',
})
```

That's why the inputs aren't really being cleared after adding books :)